src/index/blockfilterindex.h

```diff
@@ -30,6 +38,9 @@ class BlockFilterIndex final : public BaseIndex
     bool ReadFilterFromDisk(const FlatFilePos& pos, BlockFilter& filter) const;
     size_t WriteFilterToDisk(FlatFilePos& pos, const BlockFilter& filter);
 
+    Mutex m_cs_headers_cache;
+    std::unordered_map<uint256, uint256, FilterHeaderHasher> m_headers_cache GUARDED_BY(m_cs_headers_cache);
```

Comment:
If you have to change PR again, you may comment that key value is block hash and mapped value filter-header, type being the same it may be confusing at first look.

Reply:
Good point. I'll add a comment if I need to retouch the branch.